pkg/channel/message_receiver.go

```diff
@@ -178,7 +178,9 @@ func (r *MessageReceiver) ServeHTTP(response nethttp.ResponseWriter, request *ne
 		r.reporter.ReportEventCount(&args, nethttp.StatusBadRequest)
 		return
 	}
-	err = r.receiverFunc(request.Context(), channel, message, []binding.Transformer{}, utils.PassThroughHeaders(request.Header))
+	additionalHeaders := utils.PassThroughHeaders(request.Header)
+	kncloudevents.SetAcceptReplyHeader(additionalHeaders)
```

Comment:
why is the header added here? this looks like the ingress of the channel, and you are adding the header to the outbound response for the channel? I am confused.

Reply:
You are right. Removed as suggested.

Reply:
Now I remember, this was originally added to `pkg/channel/message_dispatcher.go` but slinkydeveloper suggests to move it here to avoid additional memory alloc. I think this is needed when channel is connected directly to sink.

Reply:
Just checked with an example channel configuration that the channel is connected to sink using subscription instead of being used in a MT broker, this is needed to set the header in this case. IIUC the `receiverFunc` will call the mesage_dispatcher. But I have also tried to add it in message dispatcher in a different place, this should avoid the copy of header and have the same effect of the origin code.

Reply:
Never mind, the problem was that setting the header inside the message_dispatcher will fail unit test due to race condition, so I have to either:
1. Move the logic to message receiver as before
2. Make a local copy in message dispatcher